### ISSUE TITLE
`eth_call` fails with "the tx doesn't have the correct nonce"

### ISSUE THREAD

**Issue by naddison36** [trufflesuite/ganache-cli#694]
## Expected Behavior

Calls the `eth_call` should not be validating the nonce in the transaction parameter against the current nonce.

## Current Behavior

Ganache is currently treating calls to `eth_call` like a transaction and checking the nonce in the transaction parameter against the expected next nonce. It then throws an error complaining about the nonce. An example error message is `the tx doesn't have the correct nonce. account has nonce of: 12 tx has nonce of: 11`

## Possible Solution

When `eth_call` is called, the nonce validation should be removed as it's not going to change any state. This will match the behavior of other clients.

## Steps to Reproduce (for bugs)

I'm running a Ganache server as a docker-composer service configured as follows
```
version: "2"
services:
  chain:
    image: trufflesuite/ganache-cli
    entrypoint:
      - node
      - /app/ganache-core.docker.cli.js
      - --gasLimit
      - "268435456"
      - --networkId
      - "8997"
      - --noVMErrorsOnRPCResponse
      - "--account=0x1111111111111111111111111111111111111111111111111111111111111111,300000000000000000000000"
    ports:
      - "8545:8545"
```

An example contract that will revert with a reason
```Solidity
contract Revert {
    function fail() public {
        revert("This has failed");
    }
}
```

## Context

A common pattern to get the reason code from transactions that fail from `revert` and `require` statements is to call `eth_call` with the previously failed transaction and block number of the failed transaction to get encoded reason code.

Some examples of this are
* https://gist.github.com/gluk64/fdea559472d957f1138ed93bcbc6f78a#file-reason-js
* https://github.com/pertsev/web3_utilz/blob/master/revert%20reason/index.js

Ganache can't be used as a test server if this pattern is being used. Geth, Parity and Infura work as expected.

## Your Environment
* Version used: 6.6.0
* Version of Truffle/Remix/Other tools used: Ethers.js 4.0.39
* NodeJS Version: v10.15.2
* Operating System and version (include distro if Linux): Mac OS X 10.14.6
* Link to your project or repro gist:
* Commit hash to use with above link for reproduction:
* I intend to submit a pull request to fix this issue: [ ]

Note I'm not using the latest ganache-cli version 6.7.0 as my tests which use the `ganacheProvider`, and not the JsonRpcProvider, do not work. These tests that assert against the reason codes of `revert` and `require` failures no longer work after upgrading to ganache-cli 6.7.0. Even if I explicitly specify `vmErrorsOnRPCResponse: true` as a `ganacheProvider` option, I don't get the reason codes.
